src/lspclient.rs

```diff
@@ -0,0 +1,84 @@
+use std::str::FromStr;
+
+use bitcoin::secp256k1::PublicKey;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+pub(crate) struct LspClient {
+    pub pubkey: PublicKey,
+    pub connection_string: String,
+    pub url: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct GetInfoResponse {
+    pub pubkey: String,
+    pub connection_methods: Vec<GetinfoAddress>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct GetinfoAddress {
+    #[serde(rename = "type")]
+    pub item_type: GetinfoAddressType,
+    #[serde(alias = "port")]
+    pub port: u16,
+    #[serde(alias = "address", skip_serializing_if = "Option::is_none")]
+    pub address: Option<String>,
+}
+
+/// Type of connection
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum GetinfoAddressType {
+    #[serde(rename = "dns")]
+    DNS,
+    #[serde(rename = "ipv4")]
+    IPV4,
+    #[serde(rename = "ipv6")]
+    IPV6,
+    #[serde(rename = "torv2")]
+    TORV2,
+    #[serde(rename = "torv3")]
+    TORV3,
+    #[serde(rename = "websocket")]
+    WEBSOCKET,
```

Comment:
i think there is a to_lowercase thing you can do for the whole enum rather than individually to each item